src/main/clojure/clara/rules/compiler.clj

```diff
@@ -1875,34 +1881,34 @@
-  ;; some_namespace$eval1234$one_1234
-  ;; some_namespace$eval1234$two_1235
-  ;;
-  ;; some_namespace$eval1234$one_1234 and some_namespace$eval1234$two_1235 contian the implementation of the functions,
-  ;; where some_namespace$eval1234 will contain two methods, invoke and invokeStatic.
-  ;; The invokeStatic method breaks down into something similar to a single create array call followed by 2 array set calls
-  ;; with new invocations on the 2 classes the method then returns a new vector created from the array.
-  ;;
-  ;; 5000 is lower than the absolute max to allow for modifications to how clojure compiles without needing to modify this.
-  ;; The current limit should be 5471, this is derived from the following opcode investigation:
-  ;;
-  ;; Array creation:                                                    5B
-  ;; Creating and populating the first 6 elements of the array:        60B
-  ;; Creating and populating the next 122 elements of the array:    1,342B
-  ;; Creating and populating the next 5343 elements of the array:  64,116B
-  ;; Creating the vector and the return statement:                      4B
-  ;;
-  ;; This sums to 65,527B just shy of the 65,536B method size limit.
+  "The default max number of forms that will be evaluated together as a single batch.
+   5000 is chosen here due to the way that clojure will evaluate the vector of forms extracted from the nodes.
+   The limiting factor here is the max java method size (64KiB), clojure will compile each form in the vector down into
+   its own class file and generate another class file that will reference each of the other functions and wrap them in
+   a vector inside a static method. For example,
+
+   (eval [(fn one [_] ...) (fn two [_] ...)])
+   would generate 3 classes.
+
+   some_namespace$eval1234
+   some_namespace$eval1234$one_1234
+   some_namespace$eval1234$two_1235
+
+   some_namespace$eval1234$one_1234 and some_namespace$eval1234$two_1235 contian the implementation of the functions,
+   where some_namespace$eval1234 will contain two methods, invoke and invokeStatic.
+   The invokeStatic method breaks down into something similar to a single create array call followed by 2 array set calls
+   with new invocations on the 2 classes the method then returns a new vector created from the array.
+
+   5000 is lower than the absolute max to allow for modifications to how clojure compiles without needing to modify this.
+   The current limit should be 5471, this is derived from the following opcode investigation:
+
+   Array creation:                                                    5B
+   Creating and populating the first 6 elements of the array:        60B
+   Creating and populating the next 122 elements of the array:    1,342B
+   Creating and populating the next 5343 elements of the array:  64,116B
+   Creating the vector and the return statement:                      4B
+
+   This sums to 65,527B just shy of the 65,536B method size limit."
```

Comment:
Not a blocker, but this increases the importance of setting up our tests against all relevant Clojure versions in Travis since potentially different/future Clojure versions could have differences here.  That is really something we should be doing anyway though.